File: backend/src/StyleSync.Api/Modules/Designers/Models/PortfolioItem.cs
using StyleSync.Api.Common.Shared;

namespace StyleSync.Api.Modules.Designers.Models;

public class PortfolioItem : BaseEntity
{
    public int DesignerProfileId { get; set; }
    public DesignerProfile DesignerProfile { get; set; } = default!;

    public string Title { get; set; } = default!;
    public string Description { get; set; } = default!;
    public string ImageUrl { get; set; } = default!;
    public string BudgetRangeLabel { get; set; } = default!;
    public string ClientInitials { get; set; } = default!;
    public ListingStatus CompletionStatusBadge { get; set; } = ListingStatus.Published;
}

File: flutter-app/lib/modules/designers/models/portfolio_item.dart
import 'designer_summary.dart';

class PortfolioItem {
  final int id;
  final int designerProfileId;
  final String title;
  final String description;
  final String imageUrl;
  final String budgetRangeLabel;
  final String clientInitials;
  final ListingStatus completionStatusBadge;
  final DateTime createdAtUtc;

  const PortfolioItem({
    required this.id,
    required this.designerProfileId,
    required this.title,
    required this.description,
    required this.imageUrl,
    required this.budgetRangeLabel,
    required this.clientInitials,
    required this.completionStatusBadge,
    required this.createdAtUtc,
  });

  factory PortfolioItem.fromJson(Map<String, dynamic> json) {
    return PortfolioItem(
      id: (json['id'] as num).toInt(),
      designerProfileId: (json['designerProfileId'] as num?)?.toInt() ?? 0,
      title: json['title'] as String? ?? '',
      description: json['description'] as String? ?? '',
      imageUrl: json['imageUrl'] as String? ?? '',
      budgetRangeLabel: json['budgetRangeLabel'] as String? ?? '',
      clientInitials: json['clientInitials'] as String? ?? '',
      completionStatusBadge: ListingStatus.fromValue(
          (json['completionStatusBadge'] as num?)?.toInt() ?? 1),
      createdAtUtc: json['createdAtUtc'] != null
          ? DateTime.tryParse(json['createdAtUtc'].toString()) ??
              DateTime.now()
          : DateTime.now(),
    );
  }
}
